include/boost/json/string.hpp

```diff
@@ -2653,37 +2653,101 @@ class string
 
     //------------------------------------------------------
 
+    /** Return whether the string begins with a string.
+
+        Returns `true` if the string begins with `s`, and `false` otherwise.
+
+        @par Complexity
+
+        Linear.
+
+        @param s The `string_view` to check for.
+    */
     bool
     starts_with(string_view s) const noexcept
     {
         return substr(0, s.size()) == s;
     }
 
+    /** Return whether the string begins with a character.
+
+        Returns `true` if the string begins with `ch`, and `false` otherwise.
+
+        @par Complexity
+
+        Constant.
+
+        @param ch The character to check for.
+    */
     bool
     starts_with(char ch) const noexcept
     {
         return ! empty() && front() == ch;
     }
 
+    /** Return whether the string begins with a string.
+
+        Returns `true` if the string begins with the string 
+        pointed to be `s` of length `traits_type::length(s)`,
+        and `false` otherwise.
+
+        @par Complexity
+
+        Linear.
```

Comment:
"Linear in this->size()`? or something?

Reply:
I didn't really know the notation until my last math class where we went over it. I'll be updating those to be more specific.